### DISCUSSION TITLE
Unified DSEP BAP - Enabling Access to Opportunities at Scale

### DISCUSSION THREAD

**Discussion by surendrasinghs** [beckn/DSEP-Specification#55] [Ideas]
Hello!

In the DSEP community, four tracks are under discussion 
1. Jobs, Internship, Gigs, and Volunteering
2. Scholarship, Grants and other financing options (vouchers, tokens)
3. Mentorship, Tutoring and other forms of coaching
4. Training, Courses, and other learning opportunities

What if we could have a unified end-user experience where they can discover and access opportunities from all four of the domains? 

Sharing below a few screenshots from the working document. You're invited to contribute to this. I am looking for
1. UX Designers to help visualise this even more clearly and concretely
2. App/Web/Portal/B2C end-user interface developers to help bring this to life
3. Fellow product people or anyone who wishes to co-create this and add to the imagination of the user-flow
4. Tech lead or architects to help guide developers wherever required and to ensure we build it right
5. Project manager or coordinator to keep all of us on track
6. Evangelist or business development to help us get more adopters
7. *Anyone* interested in enabling access to opportunities at scale - feel free to reach out! (User researchers, artists, marketing professionals, students or anyone)

Check out the initial draft of the envisioned **Unified BAP for Skilling, Education, and Lifelong learning**

<img width="1193" alt="image" src="https://user-images.githubusercontent.com/10580754/185110775-bd1d6750-ee60-4fb3-ad3a-0f408c1d32af.png">

<img width="1194" alt="image" src="https://user-images.githubusercontent.com/10580754/185110992-bee4821f-7266-450a-a9e5-4b802072d960.png">

<img width="1193" alt="image" src="https://user-images.githubusercontent.com/10580754/185111166-5151cf52-db85-49f7-afaf-d7efe0beaf43.png">

<img width="1192" alt="image" src="https://user-images.githubusercontent.com/10580754/185111239-4a8bfae7-f95b-400e-ba4a-20358fd67ebb.png">

<img width="1193" alt="image" src="https://user-images.githubusercontent.com/10580754/185111560-aa3be54e-82f7-4fcf-9367-98f77e8ba84a.png">

Sounds interesting? Comment below or check out [this working document](https://docs.google.com/presentation/d/1v8BizbXZ6X5Plir7gn1y1mbcQ0Re9YQR1azCFXUyGYo/edit#slide=id.p). (Request for edit/comment access to the document if necessary)

@Anand-EkStep @pramodkvarma @core-wg-admin @aks30 @RashmiG-Protean @sujeets-protean @girirajdaga @ravi-prakash-v

**Comment by pramodkvarma** [comment]
This is fantastic! Thanks @surendrasinghs. Looking forward to seeing someone volunteer for building these into a new or existing BAP apps.

**Comment by surendrasinghs** [comment]
A general progress update on this volunteer-led initiative:

The scope and milestones discussed and noted down in the presentation shared above (check slides 8 - 11).

Here's a draft user flow (UI/UX) for the job search and application flow https://www.figma.com/file/WNvcvkXUjIL3oyCjvT0pYy/Unified-DSEP-BAP?node-id=0%3A1&t=1VdIQMUh3QMGzVpC-0. This will be updated with similar discovery flow for other use-cases (Training, Mentorship, and Scholarship)

We also discussed a high-level architecture approach for the Unified BAP with @ravi-prakash-v and other volunteers. Check out https://us06web.zoom.us/rec/share/gPA7dTmjwGEvXHfbSVlI56qcCUfSCy5jvD3JfXBhkuIx7i_Q18GO8IS4CXdm0vcK.l9xVU9bvepnknRzJ 
Passcode: XV?$c=9m

@ravi-prakash-v Please share the architecture diagram used during the call.

**Comment by ravi-prakash-v** [comment]
# Unified BAP Architecture (v0.1)
Pasting the architecture diagram below. @pramodkvarma requesting your feedback on the same

![Unified BAP Architecture](https://user-images.githubusercontent.com/52468749/205589875-1eb7c34e-446b-4381-9faf-b4b057db3892.png)

#request-for-comments @beckn/dsep-working-group

**Reply by ravi-prakash-v** [reply]
CC: @scaledapps @rahulchoudhary-em

**Reply by pramodkvarma** [reply]
It looks fine at the high level. For India, Aadhaar/UPI etc will all need to be integrated quite Like Digilocker.

**Reply by ravi-prakash-v** [reply]
Noted @pramodkvarma.

@rahulchoudhary-em can you please update the architecture and publish it here?
